Riot/Modules/Rooms/SearchableDirectory/Cells/Room/DirectoryRoomTableViewCell.swift

```diff
@@ -0,0 +1,135 @@
+// 
+// Copyright 2020 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import UIKit
+import Reusable
+
+class DirectoryRoomTableViewCell: UITableViewCell {
+    
+    @IBOutlet private weak var avatarImageView: MXKImageView! {
+        didSet {
+            avatarImageView.layer.cornerRadius = avatarImageView.frame.width/2
+            avatarImageView.clipsToBounds = true
+        }
+    }
+    @IBOutlet private weak var displaynameLabel: UILabel!
+    @IBOutlet private weak var numberOfUsersLabel: UILabel!
+    @IBOutlet private weak var topicLabel: UILabel!
+    @IBOutlet private weak var joinButton: UIButton!
+    @IBOutlet private weak var joinActivityIndicator: UIActivityIndicatorView!
+
+    private weak var room: MXPublicRoom!
+    private weak var session: MXSession!
+    private var isJoined: Bool = false {
+        didSet {
+            joinButton.setTitle(isJoined ? VectorL10n.joined : VectorL10n.join, for: .normal)
+            joinButton.isUserInteractionEnabled = !isJoined
+            joinActivityIndicator.isHidden = true
+            update(theme: ThemeService.shared().theme)
+        }
+    }
+
+    func configure(withRoom room: MXPublicRoom, session: MXSession) {
```

Comment:
Dependency on MXSession is not required. We used it a lot in the past and it appeared to be a bad idea, a really bad separation of concern design.
We should pass only simple view data model.

Reply:
I've created a view model called `DirectoryRoomTableViewCellVM`, but unfortunately i had to keep the session in it, because it's needed when setting the avatar. But good news is, i've separated the join logic from the cell.